shell/edit/provisioning.cattle.io.cluster/SelectCredential.vue

```diff
@@ -103,8 +103,11 @@ export default {
 
     options() {
       const out = this.filteredCredentials.map((obj) => {
+        const duplicateFound = this.filteredCredentials.find((filtered) => filtered.nameDisplay === obj.nameDisplay);
```

Comment:
ok, my bad, i got this wrong. map must alter in place because this is now broken (for me it always returns true).

adding ` && filtered.id !== obj.id` will fix it. 

i did wonder about performance. both map and find will scale (cpu) badly with a lot of credentials.  i'm not sure how realistic that is, but would suggest something like this instead...

```
const duplicates = {};

this.filteredCredentials.forEach((cred) => {
  duplicates[cred.nameDisplay] = duplicates[cred.nameDisplay] === null ? true : null;
});

const out = this.filteredCredentials.map((obj) => ({
  // if credential name is duplicated we add the id to the label
  label: duplicates[obj.nameDisplay] ? `${ obj.nameDisplay } (${ obj.id })` : obj.nameDisplay,
  value: obj.id,
}));
```